gc/src/lib.rs

```diff
@@ -30,17 +32,13 @@ pub use gc::force_collect;
 ///
 /// See the [module level documentation](./) for more details.
 pub struct Gc<T: Trace + ?Sized + 'static> {
-    // XXX We can probably take advantage of alignment to store this
-    root: Cell<bool>,
-    _ptr: NonZero<*mut GcBox<T>>,
+    _ptr: UnsafeCell<NonZero<*mut GcBox<T>>>,
 }
 
 impl<T: ?Sized> !marker::Send for Gc<T> {}
 
 impl<T: ?Sized> !marker::Sync for Gc<T> {}
 
-impl<T: Trace + ?Sized + marker::Unsize<U>, U: Trace + ?Sized> CoerceUnsized<Gc<U>> for Gc<T> {}
```

Comment:
This PR is broken up into two commits.  The first one leaves CoerceUnsigned, but doesn't have UnsafeCell.  

I don't know how to get one without removing the other.  Maybe it can be done?  Maybe it needs support in the standard lib?